tests/test_subscriber.py

```diff
@@ -193,19 +195,160 @@ def test_on_put_invalid(self, mock_invalid_param):
+                                        "message", p.handle_event_data)
+
+    assert mock_create_indexes.called
+    assert mock_messenger.called
+
+    assert mock_create_notifications == mock_create_channel.assert_any_call(
+        "dojot.notifications", "r")
+
+    assert mock_create_all_services_devices == mock_create_channel.assert_any_call(
+        mock_config.dojot['subjects']['devices'], "r")
+
+    assert mock_create_all_services_device_data == mock_create_channel.assert_any_call(
+        mock_config.dojot['subjects']['device_data'], "r")
+
+    assert mock_on_handle_new_tenent == mock_on.assert_any_call(mock_config.dojot['subjects']['tenancy'],
+                                                                "message", p.handle_new_tenant)
+    assert mock_on_handle_notification == mock_on.assert_any_call(
+        "dojot.notifications", "message", p.handle_notification)
+
+    assert mock_on_handle_event_devices == mock_on.assert_any_call(mock_config.dojot['subjects']['devices'],
+                                                                   "message", p.handle_event_devices)
+
+    assert mock_on_handle_event_data == mock_on.assert_any_call(mock_config.dojot['subjects']['device_data'],
+                                                                "message", p.handle_event_data)
+
+    assert mock_create_channel.call_count == 3
+
+    assert mock_on.call_count == 4
+
+
+@patch.object(Persister, 'create_indexes')
+@patch.object(Config, 'load_defaults')
+@patch('history.subscriber.persister.Messenger')
+@patch.object(Messenger, 'create_channel', return_value=None)
+@patch.object(Messenger, 'on', return_value=None)
+def test_persist_only_notifications(mock_create_channel, mock_on, mock_messenger, mock_config, create_indexes):
+
+    from history.subscriber.persister import start_dojot_messenger
+    from history import conf
+    p = Persister()
+    p.create_indexes_for_notifications('admin')
+
+    mock_config.dojot = {
+        "management": {
+            "user": "dojot-management",
+            "tenant": "dojot-management"
+        },
+        "subjects": {
+            "tenancy": "dojot.tenancy",
+            "devices": "dojot.device-manager.device",
+            "device_data": "device-data"
+        }
+    }
+    # test persist only boolean valued notifications
+    conf.dojot_persist_notifications_only = True
+    start_dojot_messenger(mock_config, p)
+    mock_test_create_channel = mock_create_channel("dojot.notifications", "r")
+
+    mock_on_handle_new_tenent = mock_on(mock_config.dojot['subjects']['tenancy'],
+                                        "message", p.handle_new_tenant)
+
+    mock_on_handle_notification = mock_on(
+        "dojot.notifications", "message", p.handle_notification)
+
+    assert create_indexes.called
+    assert mock_messenger.called
+    assert mock_test_create_channel == mock_create_channel.assert_called_with(
```

Comment:
You have to test whether mock create_channel is called only once.

Reply:
fixed by modifying assert_called_once_with()